src/framework/COScript.m

```diff
@@ -405,6 +405,56 @@ - (void)deleteObjectWithName:(NSString*)name {
     [_mochaRuntime removeObjectWithName:name];
 }
 
+# pragma mark - require
+
+- (NSURL*)resolveModule:(NSString *)module currentURL:(NSURL*)currentURL {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    BOOL isRelative = [module characterAtIndex:0] == '.';
+    NSString* modulePath = [module stringByStandardizingPath];
+    NSURL* moduleURL = isRelative ? [NSURL URLWithString:modulePath relativeToURL:currentURL] : [NSURL fileURLWithPath:modulePath];
```

Comment:
Is it possible that `modulePath` could contain non or special URL characters, such as `?` or 💩 ? If so this would probably cause an unexepected URL here.

Reply:
It could, but if it does, I'm happy for this to blow up because it shouldn't

Reply:
What do you mean by "blow up" here? Is throwing an exception acceptable? I suspect it's not, and I reckon if `moduleURL` were to be `nil`, you'd get an exception.

Reply:
It's fine to throw an exception because the function is called from `require` which itself is called from a plugin script which is wrapped in a try/catch.

That being said, it might be cleaner to check if it's nil, so let's do that.

Reply:
> It's fine to throw an exception because the function is called from require which itself is called from a plugin script which is wrapped in a try/catch.

But that's not how Cocoa works. It doesn't support exceptions for flow control; they're just for programmer error. If you throw an exception, most likely you'll be left with memory that never gets deallocated, things like that.